include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1462,8 +1463,7 @@ void mergeDistSubproblems(comm::CommunicatorGrid grid,
   //
   // Note: i_split is unique
   const comm::IndexT_MPI tag = to_int(i_split);
-  applyGivensRotationsToMatrixColumns(grid.rowCommunicator(), tag, i_begin, i_end, std::move(rots),
-                                      ws.e0);
+  applyGivensRotationsToMatrixColumns(grid, row_task_chain, tag, i_begin, i_end, std::move(rots), ws.e0);
```

Comment:
Discussed this with @albestro briefly, with his argument for not hiding it inside the algorithm being that it's clear from the algorithm structure that `applyGivensRotationsToMatrixColumns` requires a sub pipeline (please correct me if I'm misrepresenting your view on this).

My counterargument is that it is not the caller's responsibility to know whether `applyGivensRotationsToMatrixColumns` requires a sub pipeline (schedules communication in subtasks) or if it suffices to pass a non-sub pipeline (all communication scheduled directly). If we call `applyGivensRotationsToMatrixColumns` with `row_task_chain.sub_pipeline()` and then change the implementation we may have taken a sub pipeline unnecessarily if we don't remember to update the call. It may be unlikely that we forget to update it, but given the two choices I don't see a clear benefit of moving the call to `sub_pipeline` out of the function.

Reply:
Decided to keep the `sub_pipeline` call inside the algorithm.